docs/conf.rst

```diff
@@ -0,0 +1,253 @@
+on how to enable this feature in the server.
+
+Open source QuestDB does not offer TLS support out of the box, but you can
+still use TLS by setting up a proxy in front of QuestDB, such as
+`HAProxy <https://www.haproxy.org/>`.
+
+* ``tls_ca`` - The remote server's certificate authority verification mechamism.
+
+  * ``'webpki_roots'``: Use the
+    `webpki-roots <https://crates.io/crates/webpki-roots>`_ Rust crate to
+    recognize certificates.
+
+  * ``'os_roots'``: Use the OS-provided certificate store.
+
+  * ``'webpki_and_os_roots'``: Use both the
+    `webpki-roots <https://crates.io/crates/webpki-roots>`_ Rust crate and
+    the OS-provided certificate store to recognize certificates.
+
+  * ``pem_file``: Path to a PEM-encoded certificate authority file.
+    This is useful for testing with self-signed certificates.
+
+  The default is: ``'webpki_roots'``.
+
+* ``tls_roots`` - ``str``: Path to a PEM-encoded certificate authority file.
+  When used it defaults the ``tls_ca`` to ``'pem_file'``.
+
+* ``tls_verify`` - ``'on'`` | ``'unsafe_off'``: Whether to verify the server's
+  certificate. This should only be used for testing as a last resort and never
+  used in production as it makes the connection vulnerable to man-in-the-middle
+  attacks.
+  
+  The default is: ``'on'``.
+
+As an example, if you are in a corporate environment and need to use the OS
+certificate store, you can use the following configuration string::
+
+    https::addr=localhost:9009;tls_ca=os_roots;
+
+Alternatively, if you are testing with a self-signed certificate, you can use
+the following configuration string::
+
+    https::addr=localhost:9009;tls_roots=/path/to/cert.pem;
+
+For more details on using self-signed test certificates, see:
+
+* For Open Source QuestDB: https://github.com/questdb/c-questdb-client/blob/main/tls_certs/README.md#self-signed-certificates
+
+* For QuestDB Enterprise: https://questdb.io/docs/operations/tls/#demo-certificates
+
+.. _sender_conf_auto_flush:
+
+Auto-flushing
+=============
+
+The following parameters control the :ref:`sender_auto_flush` behavior.
+
+* ``auto_flush`` - ``'on'`` | ``'off'``: Global switch for the auto-flushing
+  behavior.
+
+  *Default: ``'on'``.*
+
+* ``auto_flush_rows`` - ``int > 0`` | ``'off'``: The number of rows that will
+  trigger a flush. Set to ``'off'`` to disable.
+    
+  *Default: 75000 (HTTP) | 600 (TCP).*
+
+* ``auto_flush_bytes`` - ``int > 0`` | ``'off'``: The number of bytes that will
+  trigger a flush. Set to ``'off'`` to disable.
+        
+  *Default: ``'off'``.*
+
+* ``auto_flush_interval`` - ``int > 0`` | ``'off'``: The time in milliseconds
+  that will trigger a flush. Set to ``'off'`` to disable.
+    
+  *Default: 1000 (millis).*
+
+.. _sender_conf_auto_flush_interval:
+
+``auto_flush_interval``
+-----------------------
+
+It should be noted that the ``auto_flush_interval`` does not start a timer
```

Comment:
The example is clear.

However, it works from "point of confusion" into "clarity".

As an alternative mode, start from clarity and - at most - note confusion.

The following appends a brief conceptual intro to the auto-flushing section.

It then opens `auto_flush_interval` from clarity, notes contrast; demonstrates.

```
Auto-flushing
=============

Auto flushing is the automated process of periodically sending (or "flushing") the accumulated data from the client's buffer to the QuestDB database without the need for explicit commands to do so after each data entry.

The following parameters control the :ref:`sender_auto_flush` behavior.

* ``auto_flush`` - ``'on'`` | ``'off'``: Global switch for the auto-flushing
  behavior.

  *Default: ``'on'``.*

* ``auto_flush_rows`` - ``int > 0`` | ``'off'``: The number of rows that will
  trigger a flush. Set to ``'off'`` to disable.

  *Default: 75000 (HTTP) | 600 (TCP).*

* ``auto_flush_bytes`` - ``int > 0`` | ``'off'``: The number of bytes that will
  trigger a flush. Set to ``'off'`` to disable.

  *Default: ``'off'``.*

* ``auto_flush_interval`` - ``int > 0`` | ``'off'``: The time in milliseconds
  that will trigger a flush. Set to ``'off'`` to disable.

  *Default: 1000 (millis).*

.. _sender_conf_auto_flush_interval:

``auto_flush_interval``
-----------------------

The `auto_flush_interval` parameter controls how often flushing occurs. It is defined in terms of time, specifying the maximum amount of time that can pass before the buffered data is automatically flushed to the database. 

Though do note that this parameter does not create a continuous timer that  counts down each time data is added. Instead, the client checks the time elapsed since the last flush each time new data is added. If the elapsed time exceeds the specified `auto_flush_interval`, the client automatically flushes the current buffer to the database.

Consider the following example:

.. code-block:: python
    from questdb.ingress import Sender, TimestampNanos
    import time
    conf = "http::addr=localhost:9009;auto_flush_interval=1000;"
    with Sender.from_conf(conf) as sender:
        # row 1
        sender.row('table1', symbols={'sym': 'AAPL'}, at=TimestampNanos.now())
        time.sleep(60)  # sleep for 1 minute
        # row 2
        sender.row('table1', symbols={'sym': 'AAPL'}, at=TimestampNanos.now())

In this example above, "row 1" will not be flushed for a whole minute, until
"row 2" is added and the ``auto_flush_interval`` limit of 1 second is exceeded,
causing both "row 1" and "row 2" to be flushed.

If you need consistent flushing at specific intervals, you should implement your
own timer-based logic. The :ref:`sender_advanced` documentation should help you.

```

Reply:
Reworded.